### PR TITLE
Datetime input validation

### PR DESCRIPTION
**Pull request by frankfu03**

### CODE REVIEW

**Comment by lbodor** [comment]
:+1:

File: src/client/app/shared/dynamic-form-fields/abstract-gnss-controls.ts [hunk 1]
```diff
@@ -36,17 +33,13 @@ export abstract class AbstractGnssControls {
...
                     errString += 'Minimum length not reached: ' + control.errors[e].requiredLength;
                 } else if (e === 'pattern') {
-                    let pat: string;
-                    if (control.errors[e].requiredPattern === validDatetimeFormat.toString()) {
-                        pat = validDatetimeFormatHuman;
-                    } else {
-                        pat = control.errors[e].requiredPattern.toString();
-                    }
-                    errString += 'pattern required: ' + pat;
+                    errString += 'pattern required: ' + control.errors[e].requiredPattern.toString();
```
Comment (oehm-smith) [inline src/client/app/shared/dynamic-form-fields/abstract-gnss-controls.ts]: Its not going to be very readable to non-tech users.

File: src/client/app/shared/dynamic-form-fields/datetime-input.component.ts [hunk 2]
```diff
@@ -67,10 +67,20 @@ export class DatetimeInputComponent extends AbstractGnssControls implements OnIn
 
         if (this.datetimeLast !== this.datetime) {
             this.datetimeLast = this.datetime;
-            this.validateDatetime();
         }
     }
 
+    public addValidatorsToFormControl() {
```
Comment (oehm-smith) [inline src/client/app/shared/dynamic-form-fields/datetime-input.component.ts]: Good one!

File: src/client/app/shared/dynamic-form-fields/datetime-input.component.ts [hunk 3]
```diff
@@ -67,10 +67,20 @@ export class DatetimeInputComponent extends AbstractGnssControls implements OnIn
...
+        let validators: any = [];
+        if (this.required) {
+            validators.push(Validators.required);
+        }
+        validators.push(new DatetimeValidator());
+        setTimeout( () => {
+            this.formControl.setValidators(validators);
+        });
+    }
+
```
Comment (oehm-smith) [inline src/client/app/shared/dynamic-form-fields/datetime-input.component.ts]: Good one!

**Pull request merged**